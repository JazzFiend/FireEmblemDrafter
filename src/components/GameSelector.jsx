import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import Dropdown from './Dropdown/Dropdown';

class GameSelector extends PureComponent {
  static extractGameTitles(gameInfo) {
    return gameInfo.map((item) => ({
      id: item.id,
      title: item.title,
    }));
  }

  render() {
    const { draftInProgress, gameInfo, handleGameSelector } = this.props;
    const gameTitleList = GameSelector.extractGameTitles(gameInfo);
    return (
      <div>
        {
          !draftInProgress && (
            <Dropdown
              defaultText="Select a Game"
              dropdownItems={gameTitleList}
              onClick={(gameId) => handleGameSelector(gameId)}
            />
          )
        }
      </div>
    );
  }
}
export default GameSelector;

GameSelector.propTypes = {
  draftInProgress: PropTypes.bool.isRequired,
  gameInfo: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
  })).isRequired,
  handleGameSelector: PropTypes.func,
};

GameSelector.defaultProps = {
  handleGameSelector: () => {},
};
