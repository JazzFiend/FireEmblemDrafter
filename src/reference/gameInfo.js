const gameInfo = [
  {
    id: 0,
    title: 'The Sacred Stones',
    playableCharacters: [
      'Eirika', 'Ephraim', 'Seth', 'Franz', 'Gilliam', 'Vanessa', 'Moulder', 'Ross', 'Garcia', 'Neimi', 'Colm',
      'Artur', 'Lute', 'Natasha', 'Joshua', 'Forde', 'Kyle', 'Tana', 'Amelia', 'Innes', 'Gerik', 'Tethys', 'Marisa',
      'Ewan', 'Duessel', 'Cormag', 'L\'Arachel', 'Dozla', 'Saleh', 'Rennac', 'Knoll', 'Myrrh', 'Syrene',
    ],
    exclusiveCharacters: [],
    defaultTeamSize: 18,
  },
  {
    id: 1,
    title: 'The Blazing Blade',
    playableCharacters: [
      'Lyn', 'Sain', 'Kent', 'Florina', 'Wil', 'Dorcas', 'Serra', 'Erk', 'Rath', 'Matthew', 'Lucius', 'Wallace',
      'Eliwood', 'Marcus', 'Lowen', 'Rebecca', 'Bartre', 'Hector', 'Oswin', 'Guy', 'Priscilla', 'Raven', 'Canas',
      'Dart', 'Fiora', 'Legault', 'Ninian', 'Isadora', 'Heath', 'Hawkeye', 'Geitz', 'Pent', 'Louise', 'Karel',
      'Harken', 'Nino', 'Jaffar', 'Vaida', 'Renault', 'Farina', 'Karla',
    ],
    exclusiveCharacters: [
      ['Wallace', 'Geitz'],
      ['Karel', 'Harken'],
    ],
    defaultTeamSize: 15,
  },
  {
    id: 2,
    title: 'The Binding Blade',
    playableCharacters: [
      'Roy', 'Marcus', 'Alen', 'Lance', 'Wolt', 'Bors', 'Merlinus', 'Elen', 'Dieck', 'Wade', 'Lot', 'Shanna', 'Chad',
      'Lugh', 'Clarine', 'Rutger', 'Saul', 'Dorothy', 'Sue', 'Zelot', 'Trec', 'Noah', 'Astolfo', 'Lilina', 'Gwendolyn',
      'Barthe', 'Ogier', 'Fir', 'Sin', 'Gonzalez', 'Geese', 'Klein', 'Thea', 'Larum', 'Echidna', 'Elffin', 'Bartre',
      'Raigh', 'Cath', 'Melady', 'Perceval', 'Cecilia', 'Sophia', 'Igrene', 'Garret', 'Fae', 'Hugh', 'Zeiss',
      'Douglas', 'Niime', 'Juno', 'Dayan', 'Yoder', 'Karel',
    ],
    exclusiveCharacters: [
      ['Bartre', 'Echidna'],
      ['Larum', 'Elffin'],
      ['Juno', 'Dayan'],
    ],
    defaultTeamSize: 18,
  },
];

export default gameInfo;
