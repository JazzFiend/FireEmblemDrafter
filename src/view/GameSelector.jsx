import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Select from 'react-select';
import dropdownStyle from '../reference/dropdownStyle';

class GameSelector extends PureComponent {
  render() {
    const {
      defaultValue,
      gameTitleList,
      handleGameSelector,
      showGameSelector,
    } = this.props;

    return (
      <div>
        {
          showGameSelector && (
            <Select
              className="game-selector"
              classNamePrefix="game-selector"
              defaultValue={defaultValue}
              options={gameTitleList}
              onChange={handleGameSelector}
              placeholder="Game Select..."
              styles={dropdownStyle}
            />
          )
        }
      </div>
    );
  }
}
export default GameSelector;

GameSelector.propTypes = {
  defaultValue: PropTypes.shape({
    value: PropTypes.number,
    label: PropTypes.string,
  }),
  gameTitleList: PropTypes.arrayOf(PropTypes.shape({
    value: PropTypes.number,
    label: PropTypes.string,
  })).isRequired,
  handleGameSelector: PropTypes.func,
  showGameSelector: PropTypes.bool,
};

GameSelector.defaultProps = {
  defaultValue: undefined,
  handleGameSelector: () => {},
  showGameSelector: false,
};
