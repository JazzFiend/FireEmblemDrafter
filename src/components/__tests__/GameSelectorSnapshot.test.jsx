import React from 'react';
import {
  render,
  cleanup,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import GameSelector from '../GameSelector';
import gameInfo from '../../reference/gameInfo';

jest.mock('../Dropdown/Dropdown', () => () => 'Dropdown');

afterEach(cleanup);

test('renders game selection dropdown when draft is in progress', () => {
  const { asFragment } = render(
    <GameSelector
      draftInProgress={false}
      gameInfo={gameInfo}
      handleGameSelector={jest.fn()}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});

test('renders nothing when draft is not in progress', () => {
  const { asFragment } = render(
    <GameSelector
      draftInProgress={true}
      gameInfo={gameInfo}
      handleGameSelector={jest.fn()}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
