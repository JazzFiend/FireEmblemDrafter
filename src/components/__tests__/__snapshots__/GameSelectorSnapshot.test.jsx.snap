// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders game selection dropdown when draft is in progress 1`] = `
<DocumentFragment>
  <div>
    Dropdown
  </div>
</DocumentFragment>
`;

exports[`renders nothing when draft is not in progress 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
